lib/Controller/ConfigController.php

```
@@ -191,10 +193,33 @@ public function setAdminConfig(array $values): DataResponse {
 				);
 			}
 		}
-		if ((key_exists('client_id', $values) && $values['client_id'] !== $oldClientId) ||
-			(key_exists('client_secret', $values) && $values['client_secret'] !== $oldClientSecret)) {
-			$this->userManager->callForAllUsers(function (IUser $user) {
-				$this->clearUserInfo($user->getUID());
+		if (
+			// when the OP client information has changed
+			((key_exists('client_id', $values) && $values['client_id'] !== $oldClientId) ||
+			(key_exists('client_secret', $values) && $values['client_secret'] !== $oldClientSecret)) ||
+			// when the OP client information is for reset
+			($oldClientSecret && $oldClientId && $values['client_id'] === null && $values['client_secret'] === null)
+		) {
+			$this->userManager->callForAllUsers(function (IUser $user) use (
+				$oldOpenProjectOauthUrl, $oldClientId, $oldClientSecret
+			) {
+				$userUID = $user->getUID();
+				if (!$oldOpenProjectOauthUrl || !$oldClientId || !$oldClientSecret) {
+					throw new Exception('Missing required parameters.'
```

do we need to throw an exception here?
maybe better to show a message to the user and have a log entry

e.g. when the admin enters some wrong OP client id and secret at the first time setting up the app and then clicks on "Replace OpenProject OAuth values" only the message `Missing  required parameters.` is shown and the admin cannot replace the values